### PR TITLE
Display copy action when selected text is right-clicked

### LINKED ISSUE #291: Copy text message
I not found feature for right mouse button click...
 
You can fix it?

### PR DESCRIPTION
**Pull request by LorenDB**
This is maybe not the most desirable way of implementing this, because it changes the displayed "Copy" menu. Maybe I should put in a "Copy" item in the regular context menu instead?

Closes #291

### CODE REVIEW

File: resources/qml/MatrixText.qml [hunk 1]
```diff
@@ -29,8 +31,32 @@ TextEdit {
...
+
         cursorShape: parent.hoveredLink ? Qt.PointingHandCursor : Qt.ArrowCursor
+
+        onClicked: {
+            if (parent.selectedText != "")
+                copyMenu.popup();
+        }
+    }
+
+    Menu {
```
Comment (deepbluev7) [inline resources/qml/MatrixText.qml:49]: Don't create a menu per message, it is horrible for performance!

Comment (deepbluev7) [reply resources/qml/MatrixText.qml:49]: (I made that mistake before)

File: resources/qml/MatrixText.qml [hunk 2]
```diff
@@ -29,8 +31,32 @@ TextEdit {
...
+        id: copyMenu
+
+        modal: true
+
+        MenuItem {
+            id: copy
+
+            text: qsTr("Copy")
+            onTriggered: Clipboard.setText(te.selectedText);
```
Comment (deepbluev7) [inline resources/qml/MatrixText.qml:58]: https://doc.qt.io/qt-5/qml-qtquick-textedit.html#copy-method

File: resources/qml/MatrixText.qml (hunk 1 above)
Comment (LorenDB) [reply resources/qml/MatrixText.qml:49]: OK

File: resources/qml/MatrixText.qml (hunk 2 above)
Comment (LorenDB) [reply resources/qml/MatrixText.qml:58]: Who knew? :D

Comment (deepbluev7) [reply resources/qml/MatrixText.qml:58]: It was I, the great Nico! ;p

**Comment by LorenDB** [comment]
OK, I've made some changes here. The Copy item now shows up in the regular context menu, but at the expense of being able to copy selected text. Any suggestions would be appreciated.

**Comment by deepbluev7** [comment]
I would really prefer it, if you could reuse the copy function. Alternatively, you should fill the copy buffer with the plain as well as rich text.

File: src/timeline/TimelineModel.cpp [hunk 3]
```diff
@@ -773,6 +773,16 @@ TimelineModel::formatDateSeparator(QDate date) const
...
+QString
+TimelineModel::getMessageFromId(QString id) const
+{
+    auto e = events.get(id.toStdString(), "", false);
+    if (!e)
+        return "";
+    std::string body = mtx::accessors::body(*e);
+    return QString::fromStdString(body);
+}
+
```
Comment (deepbluev7) [inline src/timeline/TimelineModel.cpp:785]: We already have getDump(eventId), which you can call and then access body and such as you like in Qml.

**Comment by deepbluev7** [comment]
I found an easier way to implement this while respecting selections, so I'm closing this. Thank you though!

**Comment by LorenDB** [comment]
No problem, I'd run out of inspiration anyway.

**Pull request closed without merging**